### PR TITLE
Run precommit before gradle check

### PR DESCRIPTION
**Pull request by owaiskazi19**
Signed-off-by: Owais Kazi [owaiskazi19@gmail.com](mailto:owaiskazi19@gmail.com)

### Description
This PR adds dependency of precommit to run successfully before running gradle check. Thus saving time and collecting errors before waiting for the entire gradle check suite to run.

Dependency chart: spotless -> precommit -> gradle check

For more information: https://github.com/opensearch-project/OpenSearch/pull/5027 and https://github.com/opensearch-project/OpenSearch/pull/4679

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/gradle-check.yml [hunk 1]
```diff
@@ -9,12 +9,37 @@ on:
...
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: 11
+          distribution: adopt
```
Comment (reta) [inline .github/workflows/gradle-check.yml:32]: `distribution: temurin` please

**Review by dblock — CHANGES_REQUESTED** [review]
Change the JDK distribution to keep it consistent.

File: .github/workflows/gradle-check.yml (hunk 1 above)
Comment (owaiskazi19) [reply .github/workflows/gradle-check.yml]: I copy/pasted precommit from [main](https://github.com/opensearch-project/OpenSearch/blob/main/.github/workflows/precommit.yml#L16) which has `adopt` distribution. Updated it.

**Comment by owaiskazi19** [comment]
Github Action worked fine on my [fork](https://github.com/owaiskazi19/OpenSearch/pull/220). 
The sequence was spotless -> precommit -> gradle check. Not sure why it's not working the same way here.
cc: @reta

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
> Github Action worked fine on my [fork](https://github.com/owaiskazi19/OpenSearch/pull/220). The sequence was spotless -> precommit -> gradle check. Not sure why it's not working the same way here. cc: @reta @peterzhuamazon

Hm .. it seems like the actions are talking from `main` branch, may be because the branch is protected? (and it is not in your fork)

**Review by dblock — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
@owaiskazi19 please backport this to all the branches.
Including all the feature branch as well.

Thanks.

**Comment by owaiskazi19** [comment]
> This PR has caused a lot of issue and is not even checking out the correct commit to run spotless and gradle precommit: See:
> 
> * [Bump nebula-publishing-plugin from v4.4.0 to v4.6.0. #5127 (comment)](https://github.com/opensearch-project/OpenSearch/pull/5127#issuecomment-1306588426)

We were under the impression it's because of the branch protection rule. What are the correct commits to run spotless and precommit @peterzhuamazon ?

**Pull request merged**